Various minor improvements and stylistic changes

StephanTLavavej opened the pull request:
These changes are organized into a dozen separate commits which are independent (although some of them touch the same code). I've verified that x86/x64 codegen is unaffected, except that "Avoid cast within multipleOfPowerOf5()." uses a different comparison instruction (no change to code size).

ulfjack commented:
I'm currently traveling, and this is quite a bit to review. Will do when I'm back home.

StephanTLavavej commented:
Rebased and added more changes:
* More preincrement/predecrement and more braces (in preprocessor branches that I skipped before).
* Added const to variables/parameters. (I believe I verified through Wandbox that the HAS_UINT128 branch compiles with this change.)
* Changed calloc to malloc for a theoretical efficiency improvement.
